src/core/package-json.spec.ts

```diff
@@ -23,4 +26,36 @@ describe('determineLanguageWithVersionUsed', () => {
   
       expect(await determineLanguagesWithVersionUsed(packageJsonMap)).toEqual(['angular-7.0.0', 'react-16.7.0', 'vue-15.0.0']);
     });
-  });
\ No newline at end of file
+  });
+
+jest.mock('fs');
+jest.mock('path');
+
+describe('searchForPackageJson', () => {
+  it('should return the path to the package.json file', () => {
+    (fs.readdirSync as jest.Mock).mockReturnValueOnce(['package.json', 'src', 'public']);
```

Comment:
for this unit test the do no mock out readdirsync. Instead call the function directly with it's full functionality as expected

Reply:
create a mock package.json file and use that for the unit test